galleries/users_explain/index.rst

```diff
@@ -0,0 +1,21 @@
+.. _users-guide-explain:
+.. _users-guide-using:
+
+.. redirect-from:: /users/explain
+
+Using Matplotlib
```

Comment:
Before https://matplotlib.org/stable/users/explain/index.html titled "Explanations". Now this URL titles "Using Matplotlib", which is both confusing and changing the semantics.

Coming from
![image](https://user-images.githubusercontent.com/2836374/225961084-2c4dbb5f-ac1e-47fb-92fd-0ead1deada3f.png)

I categorize "Using matplotlib" in the top half "partical steps", whereas "Explanations" is in the lower part. While we can still move content around, we should get the organizational structure clear in this PR.

Reply:
I think that graphic is a useful guide.  OTOH I don't agree that any given document should be clearly in one section or another or that the top-level doc organization should strictly follow this.    Imagine a textbook that only had theory in half the chapters, and worked examples in the other half.   A tutorial with no basic "theory" isn't much use, similarly just discussing things in abstract is rarely a good way to learn.  

What I have tried to do here is high-level Explanation of what an important part of the library is (say Axes), mixed with simple examples (mini Tutorials).  Further _Explanations_ (what we had in "Intermediate" and "Advanced") are then in the table of contents of that Section, and we should add more links to the examples (eg, _Tutorials_/_How-to guides_).  

My argument would be that we never had the high-level explanations.  Sometimes the API _Reference_ had some explanation, and we had some getting-started tutorials, but we never explained what a Figure was or what an Axes was.  Most of the so-called tutorials were really long Explanations that were easier to write in sphinx-gallery than rst.  Some of the material in users/explain was pretty esoteric, other things were really fundamental.  

So, I'm open to changing this organization, or changing the name "Using Matplotlib", but I think this is already substantially better than before, and can be filled out over the next few months to be really good.